core/embed/rust/src/ui/model_mercury/component/button.rs

```diff
@@ -525,46 +527,39 @@ impl IconText {
         baseline_offset: Offset,
         alpha: u8,
     ) {
-        let width = self.text.map(|t| style.font.text_width(t));
+        let mut show_text = |text: &str, rect: Rect| {
+            let text_pos = rect.left_center() + baseline_offset;
+            let text_pos = Point::new(rect.top_left().x + Self::ICON_SPACE, text_pos.y);
+            shape::Text::new(text_pos, text)
+                .with_font(style.font)
+                .with_fg(style.text_color)
+                .with_alpha(alpha)
+                .render(target)
+        };
 
-        let mut use_icon = false;
-        let mut use_text = false;
+        self.text.map(|t| {
+            let (t1, t2) = split_two_lines(
+                t,
+                style.font,
+                area.width() - Self::ICON_SPACE - Self::TEXT_MARGIN,
+            );
 
-        let mut icon_pos = Point::new(
+            if t1.is_empty() || t2.is_empty() {
+                show_text(if t1.is_empty() { t2 } else { t1 }, area);
```

Comment:
pretty sure this will work?
```suggestion
                show_text(t, area);
```